workflows/iuf/operations/management-nodes-rollout/management-storage-nodes-rollout.yaml

```
@@ -112,9 +112,19 @@ spec:
                   value: |
                     echo "INFO This stage will start a new Argo workflow with a name containing 'ncn-lifecycle-rebuild'."
                     echo '{{inputs.parameters.global_params}}' > global.params.data
-                    prepare_images_ouput=$(cat global.params.data | jq -r '.stage_params["prepare-images"].global["prepare-management-images"]["sat-bootprep-run"].script_stdout')
-                    image=$(echo $prepare_images_ouput | jq '.images | map(select(.configuration_group_names[] == "Management_Storage")) | .[].final_image_id' | tr -d '"')
-                    configuration=$(echo $prepare_images_ouput | jq '.images | map(select(.configuration_group_names[] == "Management_Storage")) | .[].configuration' | tr -d '"')
+
+                    cfs_config="{{=jsonpath(inputs.parameters.global_params,'$.input_params.cfs_configuration')}}"
+                    ims_image="{{=jsonpath(inputs.parameters.global_params,'$.input_params.boot_image_management')}}"
+                    if [[ -n $cfs_config && -n $ims_image ]]; then
+                      echo "INFO Rebuilding $rebuild_set using passed parameters $ims_image and $cfs_config"
+                      image=$ims_image
+                      configuration=$cfs_config
+                    else
+                      echo "INFO Starting rebuild of $rebuild_set using image built in the prepare images stage"
+                      prepare_images_ouput=$(cat global.params.data | jq -r '.stage_params["prepare-images"].global["prepare-management-images"]["sat-bootprep-run"].script_stdout')
+                      image=$(echo $prepare_images_ouput | jq '.images | map(select(.configuration_group_names[] == "Management_Storage")) | .[].final_image_id' | tr -d '"')
+                      configuration=$(echo $prepare_images_ouput | jq '.images | map(select(.configuration_group_names[] == "Management_Storage")) | .[].configuration' | tr -d '"')
```

Use `jq -r`, not `tr -d`
```suggestion
                      image=$(echo $prepare_images_ouput | jq -r '.images | map(select(.configuration_group_names[] == "Management_Storage")) | .[].final_image_id')
                      configuration=$(echo $prepare_images_ouput | jq -r '.images | map(select(.configuration_group_names[] == "Management_Storage")) | .[].configuration')
```